### PR TITLE
cellSysutilAvc2: improve unload and some more stuff

### PR DESCRIPTION
**Pull request by Megamouse**
- various improvements in cellSysutilAvc2

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 1]
```diff
@@ -94,11 +99,12 @@ struct avc2_settings
 		save(ar);
 	}
 
-	void save(utils::serial& ar)
+	void save(utils::serial& ar) const
 	{
 		GET_OR_USE_SERIALIZATION_VERSION(ar.is_writing(), cellSysutil);
 
-		ar(avc2_cb, avc2_cb_arg, streaming_mode, mic_out_stream_sharing, video_stream_sharing, total_video_bitrate);
+		ar(avc2_cb, avc2_cb_arg, streaming_mode, mic_out_stream_sharing, video_stream_sharing, total_video_bitrate,
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:106]: @elad335 is there anything else to do for savestates here?

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:106]: Add a savestate module version check for these components.

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 2]
```diff
@@ -94,11 +99,12 @@ struct avc2_settings
 		save(ar);
 	}
 
-	void save(utils::serial& ar)
+	void save(utils::serial& ar) const
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:102]: not const

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp]: Done, I think

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 3]
```diff
@@ -96,9 +101,14 @@ struct avc2_settings
 
 	void save(utils::serial& ar)
 	{
-		GET_OR_USE_SERIALIZATION_VERSION(ar.is_writing(), cellSysutil);
+		[[maybe_unused]] const s32 version = GET_OR_USE_SERIALIZATION_VERSION(ar.is_writing(), cellSysutil);
 
 		ar(avc2_cb, avc2_cb_arg, streaming_mode, mic_out_stream_sharing, video_stream_sharing, total_video_bitrate);
+
+		if (version >= 2)
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:108]:
```suggestion
		if (ar.is_writing() || version >= 2)
```

**Review by elad335 — CHANGES_REQUESTED** [review]

**Review by elad335 — APPROVED** [review]

**Pull request merged**